### PR TITLE
Add an option to marker/has line symbol layers to control whether the first/last vertex option should respect multipart geometries

### PR DESCRIPTION
**Pull request by nyalldawson**
The default is not to respect these, so first and last vertices are symbolized for every part of a multi-part geometry (this matches
the old behaviour). By opting in to the "respect multipart geometries" option, the symbols will only be placed on the VERY first or very LAST vertex in the whole multi-part geometry.

Sponsored by North Road, thanks to SLYR

### CODE REVIEW

File: src/core/symbology/qgsfillsymbollayer.cpp [hunk 1]
```diff
@@ -4397,9 +4397,6 @@ QColor QgsCentroidFillSymbolLayer::color() const
 void QgsCentroidFillSymbolLayer::startRender( QgsSymbolRenderContext &context )
 {
   mMarker->startRender( context.renderContext(), context.fields() );
-
-  mCurrentFeatureId = -1;
```
Comment (nirvn) [inline src/core/symbology/qgsfillsymbollayer.cpp:4401]: That's deliberate?

**Review by nirvn — APPROVED** [review]

File: src/core/symbology/qgsfillsymbollayer.cpp (hunk 1 above)
Comment (nyalldawson) [reply src/core/symbology/qgsfillsymbollayer.cpp:4401]: Yeah, it's not used anywhere. I just noticed it while using that other class as a model.

Comment (nirvn) [reply src/core/symbology/qgsfillsymbollayer.cpp:4401]: @nyalldawson , OK, well, merge at will :)

File: src/core/symbology/qgslinesymbollayer.h [hunk 2]
```diff
@@ -651,6 +651,38 @@ class CORE_EXPORT QgsTemplatedLineSymbolLayerBase : public QgsLineSymbolLayer
...
+     * the first/last vertex of EVERY part of a multipart feature.
+     *
+     * If TRUE, then Qgis::MarkerLinePlacement::FirstVertex or
+     * Qgis::MarkerLinePlacement::LastVertex placements will result in a symbol on
+     * the first/last vertex of the overall multipart geometry only.
+     *
+     * \see setRespectMultipart()
+     * \since QGIS 3.24
+     */
+    bool respectMultipart() const { return mRespectMultipart; }
```
Comment (3nids) [inline src/core/symbology/qgslinesymbollayer.h:668]: I'm not a super fan of this wording. What about `placeOnEveryPart` or something similar?

**Pull request merged**